### PR TITLE
[packaging] Publish xds-protos as part of the standard package pipeline

### PR DESCRIPTION
**Pull request by gnossen**
This PR:

- Fixes the xds-protos Python package, which was broken when the `udpa` submodule was removed
  - This required re-adding the protoc-gen-validate submodule
- Adds non-Bazel tests for xds-protos and all of its dependent packages
- Versions xds-protos the same way as the rest of the Python packages
- Fixes Python 3.11 support in `run_tests.py`, which is necessary for the testing mentioned above

CC @sergiitk You won't be able to consume this in the interop tests until it makes it into a release. I'm thinking I'll want to backport this to the 1.57.x branch to make that happen faster.

CC @drfloob to inform him about the likely backport.

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -729,35 +728,54 @@ def configure(self, config, args):
...
+            if os.name != "nt":
+                jobs.append(
+                    self.config.job_spec(
+                        [
+                            python_config.python_path,
+                            "tools/distrib/python/xds_protos/generated_file_import_test.py",
+                        ],
+                        timeout_seconds=60,
+                        environ=_FORCE_ENVIRON_FOR_WRAPPERS,
+                        shortname="{}.xds_protos".format(python_config.name),
```
Comment (sergiitk) [inline tools/run_tests/run_tests.py:744]:
```suggestion
                        shortname=f"{python_config.name}.xds_protos",
```

File: tools/run_tests/run_tests.py [hunk 2]
```diff
@@ -729,35 +728,54 @@ def configure(self, config, args):
...
+                                + [self._TEST_COMMAND[io_platform]],
+                                timeout_seconds=8 * 60,
+                                environ=dict(
+                                    GRPC_PYTHON_TESTRUNNER_FILTER=str(
+                                        test_case
+                                    ),
+                                    **environment,
+                                ),
+                                shortname="%s.%s.%s"
+                                % (python_config.name, io_platform, test_case),
```
Comment (sergiitk) [inline tools/run_tests/run_tests.py:774]:
```suggestion
                                shortname=f"{python_config.name}.{io_platform}.{test_case}",
```

**Review by sergiitk — APPROVED** [review]
Minor nits and some typo fixes

File: src/python/grpcio_csds/setup.py [hunk 3]
```diff
@@ -40,7 +40,7 @@
 
 INSTALL_REQUIRES = (
     "protobuf>=4.21.6",
-    "xds-protos>=0.0.7",
+    "xds-protos=={version}".format(version=grpc_version.VERSION),
```
[the remark this answers is not in the document]
Comment (gnossen) [reply src/python/grpcio_csds/setup.py]: Done.

File: templates/tools/distrib/python/xds_protos/grpc_version.py.template [hunk 4]
```diff
@@ -0,0 +1,20 @@
+%YAML 1.2
+--- |
+  # Copyright 2020 gRPC authors.
```
[the remark this answers is not in the document]
Comment (gnossen) [reply templates/tools/distrib/python/xds_protos/grpc_version.py.template]: Done.

File: templates/tools/dockerfile/compile_python_311.include [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#=================
+# Compile CPython 3.:1.4 from source
```
[the remark this answers is not in the document]
Comment (gnossen) [reply templates/tools/dockerfile/compile_python_311.include]: Done.

File: tools/distrib/python/xds_protos/build.py [hunk 6]
```diff
@@ -19,19 +19,27 @@
...
-    "envoy/service/metrics/v2",
-    "envoy/service/metrics/v3",
-    "envoy/service/metrics/v4alpha",
+    localize_path(p)
+    for p in [
+        # Requires extra dependency to Prometheus protos
+        "envoy/service/metrics/v2",
+        "envoy/service/metrics/v3",
+        "envoy/service/metrics/v4alpha",
+    ]
```
[the remark this answers is not in the document]
Comment (gnossen) [reply tools/distrib/python/xds_protos/build.py]: Done.

File: tools/run_tests/run_tests.py (hunk 1 above)
Comment (gnossen) [reply tools/run_tests/run_tests.py]: Done.

File: tools/run_tests/run_tests.py (hunk 2 above)
Comment (gnossen) [reply tools/run_tests/run_tests.py]: Done.

**Review by sergiitk — APPROVED** [review]

**Comment by gnossen** [comment]
CC @xuanwang-amos

**Pull request merged**